### PR TITLE
[RFC] lisa.tests: load_tracking: Invariance: Treat load like util

### PR DESCRIPTION
**Pull request by douglas-raillard-arm**
Do not rescale the signal specifically when it is the load.

### CODE REVIEW

File: lisa/tests/scheduler/load_tracking.py [hunk 1]
```diff
@@ -309,9 +309,8 @@ def _test_task_signal(self, signal_name, allowed_error_pct,
 
         signal_mean = area_under_curve(signal) / (signal.index[-1] - signal.index[0])
 
-        if signal_name == 'load':
-            # Load isn't CPU invariant
-            exp_signal /= (self.plat_info['cpu-capacities'][cpu] / UTIL_SCALE)
+        # Since load is now CPU invariant, we don't rescale it back
```
Comment (valschneider) [inline lisa/tests/scheduler/load_tracking.py]: Don't we want to guard this against a kernel version check, or did I misunderstand the last discussion we had about this?

At the very least we could have a `logger.warning()` that warns about kernel version < 5.0

Comment (douglas-raillard-arm) [reply lisa/tests/scheduler/load_tracking.py]: @ionela-voinescu was more keen on just supporting the current mainline kernel. I don't mind either.

Comment (ionela-voinescu) [reply lisa/tests/scheduler/load_tracking.py]: I don't believe checking for the kernel version is always safe. It's possible people might backport these patches to older ( < 5.0 ) android kernels so at least controlling test functionality (cpu scaled or not scaled load) based on kernel version seems wrong to me. But I doubt newer kernels would be missing this functionality so it seems right to me to warn people that use older kernels that our estimation of load can be wrong.

Comment (douglas-raillard-arm) [reply lisa/tests/scheduler/load_tracking.py]: Ok, let's put a warning if version < 5.0 and unconditionally use the new computation

Comment (qperret) [reply lisa/tests/scheduler/load_tracking.py]: This was introduced in 5.1 I believe

**Comment by douglas-raillard-arm** [comment]
It will now log it at the warning level for kernels < 5.1

File: lisa/tests/scheduler/load_tracking.py [hunk 2]
```diff
@@ -309,9 +309,18 @@ def _test_task_signal(self, signal_name, allowed_error_pct,
...
-            # Load isn't CPU invariant
-            exp_signal /= (self.plat_info['cpu-capacities'][cpu] / UTIL_SCALE)
+        # Since load is now CPU invariant in recent kernel versions, we don't
+        # rescale it back. To match the old behavior, that line is
+        # needed:
+        #  exp_signal /= (self.plat_info['cpu-capacities'][cpu] / UTIL_SCALE)
+        kernel_version = self.plat_info['kernel']['version']
+        if (
+            signal_name == 'load'
+            and kernel_version.parts < (5, 1, 0)
```
Comment (qperret) [inline lisa/tests/scheduler/load_tracking.py]: Just wondering, how does that work with rc versions ? Is 5.1-rc1 < 5.1.0 ?

Comment (douglas-raillard-arm) [reply lisa/tests/scheduler/load_tracking.py]: Good point, "rc" is not in `.parts` tuple, but will lead to `minor=None`, so the comparison will raise an exception if we get to that. Instead, I can replace it with `kernel_version.parts[:2] < (5, 1)` to just ignore the rc and minor numbers (comparing tuples with different number of parts works but seems to give surprising results wrt equality).

**Comment by douglas-raillard-arm** [comment]
Updated to ignore minor version number and rc number (i.e. in 5.1.2, only 5.1 will be taken into account)

**Pull request merged**